modules/python3/src/volumepy.cpp

```diff
@@ -0,0 +1,177 @@
+
+        auto shape = extent == 1 ? pybind11::array::ShapeContainer{dims.x, dims.y, dims.z}
+                                 : pybind11::array::ShapeContainer{dims.x, dims.y, dims.z, extent};
+        pybind11::array_t<CompType> data{shape};
+
+        if (pybind11::array::c_style == (data.flags() & pybind11::array::c_style)) {
+            std::memcpy(data.mutable_data(0), vr->getData(), data.nbytes());
+        } else {
+            throw Exception("Unable to convert from VolumePy to VolumeRAM",
+                            IVW_CONTEXT_CUSTOM("VolumeRAM2PyConverter"));
+        }
+
+        return data;
+    });
+
+    auto volumeDst = std::make_shared<VolumePy>(
+        data, volumeSrc->getSwizzleMask(), volumeSrc->getInterpolation(), volumeSrc->getWrapping());
+    return volumeDst;
+}
+
+void VolumeRAM2PyConverter::update(std::shared_ptr<const VolumeRAM> volumeSrc,
+                                   std::shared_ptr<VolumePy> volumeDst) const {
+    volumeDst->setDimensions(volumeSrc->getDimensions());
+    volumeDst->setSwizzleMask(volumeSrc->getSwizzleMask());
+    volumeDst->setInterpolation(volumeSrc->getInterpolation());
+    volumeDst->setWrapping(volumeSrc->getWrapping());
+
+    if (pybind11::array::c_style == (volumeDst->data().flags() & pybind11::array::c_style)) {
+        std::memcpy(volumeDst->data().mutable_data(0), volumeSrc->getData(),
+                    volumeDst->data().nbytes());
+    } else {
+        throw Exception("Unable to convert from VolumePy to VolumeRAM", IVW_CONTEXT);
+    }
+}
+
+std::shared_ptr<VolumeRAM> VolumePy2RAMConverter::createFrom(
+    std::shared_ptr<const VolumePy> volumeSrc) const {
+
+    auto volumeDst = createVolumeRAM(volumeSrc->getDimensions(), volumeSrc->getDataFormat(),
+                                     nullptr, volumeSrc->getSwizzleMask(),
+                                     volumeSrc->getInterpolation(), volumeSrc->getWrapping());
+
+    auto dst = volumeDst->getData();
+    auto src = volumeSrc->data().data(0);
+    auto size = volumeSrc->data().nbytes();
+
+    if (pybind11::array::c_style == (volumeSrc->data().flags() & pybind11::array::c_style)) {
+        std::memcpy(dst, src, size);
+    } else {
+        throw Exception("Unable to convert from VolumePy to VolumeRAM", IVW_CONTEXT);
+    }
+
+    return volumeDst;
+}
+
+void VolumePy2RAMConverter::update(std::shared_ptr<const VolumePy> volumeSrc,
+                                   std::shared_ptr<VolumeRAM> volumeDst) const {
+    volumeDst->setDimensions(volumeSrc->getDimensions());
+    volumeDst->setSwizzleMask(volumeSrc->getSwizzleMask());
+    volumeDst->setInterpolation(volumeSrc->getInterpolation());
+    volumeDst->setWrapping(volumeSrc->getWrapping());
+
+    auto dst = volumeDst->getData();
+    auto src = volumeSrc->data().data(0);
+    auto size = volumeSrc->data().nbytes();
+
+    if (pybind11::array::c_style == (volumeSrc->data().flags() & pybind11::array::c_style)) {
+        std::memcpy(dst, src, size);
```

Comment:
Would have been nice if could share the data, but I understand that it might not be possible at the moment

Reply:
share with what? VolumeRAM?

Reply:
Yes, in a similar fashion as OpenGL and OpenCL can share data

Reply:
Might, be possible, although the sharing done with OpenGL and OpenCL is also very very shady.

Reply:
Will be in the future in such case.